### PR TITLE
Compiler warnings on various platforms

### PR DESCRIPTION
**Pull request by garbear**
Alongside #12238 I'm submitting my precious cache of warnings fixes and improvements that I've been hoarding over the last few months.

### CODE REVIEW

File: xbmc/guilib/GUITexture.cpp [hunk 1]
```diff
@@ -274,8 +274,8 @@ void CGUITextureBase::Render(float left, float top, float right, float bottom, f
   y[3] = ROUND_TO_PIXEL(g_graphicsContext.ScaleFinalYCoord(vertex.x1, vertex.y2));
   z[3] = ROUND_TO_PIXEL(g_graphicsContext.ScaleFinalZCoord(vertex.x1, vertex.y2));
 
-  if (y[2] == y[0]) y[2] += 1.0f; if (x[2] == x[0]) x[2] += 1.0f;
-  if (y[3] == y[1]) y[3] += 1.0f; if (x[3] == x[1]) x[3] += 1.0f;
+  if (y[2] == y[0]) { y[2] += 1.0f; } if (x[2] == x[0]) { x[2] += 1.0f; }
+  if (y[3] == y[1]) { y[3] += 1.0f; } if (x[3] == x[1]) { x[3] += 1.0f; }
```
Comment (Rechi) [inline xbmc/guilib/GUITexture.cpp]: split this into multiple lines, then the curly brackets shouldn't be needed.

Comment (garbear) [inline xbmc/guilib/GUITexture.cpp]: fixed

**Comment by MartijnKaijser** [comment]
jenkins build this please

**Comment by MartijnKaijser** [comment]
jenkins build this please

**Pull request merged**